exporter/metrics_exporter.go

```diff
@@ -141,20 +147,45 @@ func (m *metricsExporter) recordMetrics() {
 		// concurrently, so we assume that as long as that cap isn't enormous (which
 		// would only occur on an enormous cluster), it's practical to iterate over
 		// all the running workers.
+		var countError error
 		if phase == core.WorkerPhaseRunning {
-			var pendingJobs int
-			// TODO: This logic doesn't currently account for the possibility that
-			// there are more pages of events with running workers, so this count
-			// of pending jobs could be wrong.
-			for _, event := range events.Items {
-				for _, job := range event.Worker.Jobs {
-					if job.Status.Phase == core.JobPhasePending {
-						pendingJobs++
-					}
+			pendingJobs := countJobStatus(events, core.JobPhasePending)
+			for events.Continue != "" {
```

Comment:
I think this whole algorithm ends up simpler if you use a post-test loop instead of a pre-test loop.

Pseudocode:

```
var pendingJobs int
for {
  for event in events.items {
    for job in event.jobs {
      if job.pending {
        pendingJobs++
      }
    }
  }
  if events.continue == "" {
    break
  }
  events, err = apiClient.core.events.list(options: {continue: event.continue})
  if err {
    log(err)
    break
  }
}
```

Reply:
You're right, this makes a lot more sense. Making the change now!